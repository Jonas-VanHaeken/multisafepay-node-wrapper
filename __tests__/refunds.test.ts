import axios from 'axios';
import Refunds from '../src/resources/refunds';
import createRefundMock from '../__mocks__/createRefund.mock.json';
import createRefundResponse from '../__stubs__/createRefund.stub.json';

import MockAdapter from 'axios-mock-adapter';

const mock = new MockAdapter(axios);

describe('refunds', () => {
  let refunds: any;
  beforeEach(() => {
    refunds = new Refunds(axios.create());
  });

  it('create refund', async () => {
    mock
      .onPost('/orders/my-order-id-1/refunds', createRefundMock)
      .reply(200, createRefundResponse);

    const resp = await refunds.create('my-order-id-1', createRefundMock);

    expect(resp).toMatchSnapshot();
  });

  it('create refund with error', async () => {
    const error = {
      error_code: 404,
      error_info: 'Not found',
    };
    mock
      .onPost('/orders/my-order-id-1a/refunds', createRefundMock)
      .reply(404, error);

    const resp = await refunds.create('my-order-id-1a', createRefundMock);

    expect(resp).toEqual(error);
  });
});
