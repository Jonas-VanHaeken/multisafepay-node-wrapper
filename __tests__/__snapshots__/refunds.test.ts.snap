// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`refunds create refund 1`] = `
Object {
  "data": Object {
    "refund_id": 3326969,
    "transaction_id": 3326965,
  },
  "success": true,
}
`;
